browser-switch/src/main/java/com/braintreepayments/api/BrowserSwitchRequest.java

```diff
@@ -48,24 +66,20 @@ JSONObject getMetadata() {
         return metadata;
     }
 
-    boolean getShouldNotifyCancellation() {
-        return shouldNotifyCancellation;
-    }
-
-    void setShouldNotifyCancellation(boolean shouldNotifyCancellation) {
-        this.shouldNotifyCancellation = shouldNotifyCancellation;
-    }
+    @NonNull
+    String toBase64EncodedJSON() throws BrowserSwitchException {
+        try {
+            JSONObject requestJSON = new JSONObject()
+                    .put(KEY_REQUEST_CODE, requestCode)
+                    .put(KEY_URL, url.toString())
+                    .put(KEY_RETURN_URL_SCHEME, returnUrlScheme)
+                    .putOpt(KEY_METADATA, metadata);
 
-    String toJson() throws JSONException {
-        JSONObject result = new JSONObject();
-        result.put("requestCode", requestCode);
-        result.put("url", url.toString());
-        result.put("returnUrlScheme", returnUrlScheme);
-        result.put("shouldNotify", shouldNotifyCancellation);
-        if (metadata != null) {
-            result.put("metadata", metadata);
+            byte[] requestJSONBytes = requestJSON.toString().getBytes(StandardCharsets.UTF_8);
```

Comment:
Base 64 encoding is technically optional, though it does make the state string appear to the merchant as an [opaque data type](https://en.wikipedia.org/wiki/Opaque_data_type) to discourage tampering. Base64 encoding also has an added benefit of [masking the data](https://security.stackexchange.com/questions/203664/whats-the-purpose-of-base64-if-knowing-its-base64-makes-it-easy-to-decode) to prevent casual observation of browser switch metadata e.g. intercepting an ec-token over a screen share.